library/alloc/src/collections/binary_heap/mod.rs

```diff
@@ -1178,10 +1237,17 @@ impl<T> BinaryHeap<T> {
     /// ```
     #[must_use = "`self` will be dropped if the result is not used"]
     #[stable(feature = "binary_heap_extras_15", since = "1.5.0")]
-    pub fn into_vec(self) -> Vec<T> {
+    pub fn into_vec(self) -> Vec<T, A> {
         self.into()
     }
 
+    /// Returns a reference to the underlying allocator.
+    #[unstable(feature = "allocator_api", issue = "32838")]
+    #[inline]
+    pub fn allocator(&self) -> &A {
+        self.data.allocator()
+    }
```

Comment:
The iterators should have a method like this too.

Reply:
I disagree, there's no reason for the iterator to care about the allocator. None of the iterators on Vec or HashMap have such a method.

Reply:
https://doc.rust-lang.org/nightly/std/vec/struct.IntoIter.html#method.allocator
https://doc.rust-lang.org/nightly/std/vec/struct.Drain.html#method.allocator

Reply:
Oh, I didn't realize we had added those. My bad.

Reply:
Added to IntoIter, IntoIterSorted, Drain and DrainSorted. I left PeekMut out of if for now, but let me know, and I can add it too.